kong/tools/channel.lua

```diff
@@ -0,0 +1,158 @@
+
+local min = math.min
+local max = math.max
+
+local now = ngx.now
+local sleep = ngx.sleep
+
+local DEFAULT_EXPTIME = 3600
+local DEFAULT_TIMEOUT = 5
+local NAME_KEY = "channel_up"
+local POST_VAL_KEY_PREFIX = "channel_post_value_"
+local RESP_VAL_KEY_PREFIX = "channel_resp_value_"
+
+
+local function waitstep(step, deadline)
+  sleep(step)
+  return min(max(0.001, step * 2), deadline-now(), 0.5)
+end
+
+
+local function add_wait(dict, key, val, exptime, deadline)
```

Comment:
WDYT about naming these `*_wait` functions to give a better description of what they are doing as it was not immediately obvious until looking through the function?

```suggestion
local function add_key_value_wait(dict, key, val, exptime, deadline)
```

I am not even a fan of this name `add_key_value_wait`; naming is hard.

Reply:
yeah, naming is hard.  basically these are the same as the shdict `:add()` and `:get()` methods but which block the coroutine until the operation gets done (or timeout).  i'll add some doccoments there.